pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +270,71 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+	partitionList, err := t.GetUniquePartitionList(txn)
+	if err != nil {
+		return resources, stats, errors.Wrapf(err, "failed to get partition list from table:%v", t.tableName)
+	}
+
+	tablePrefix := "/" + t.tableName + "/"
+	iterOpt := badger.DefaultIteratorOptions
+	iterOpt.Prefix = []byte(tablePrefix)
+	itr := txn.NewIterator(iterOpt)
+	defer itr.Close()
+
+	startPartition := untyped.GetPartitionId(startTime)
+	endPartition := untyped.GetPartitionId(endTime)
+	lastPartition := ""
+	for i := len(partitionList) - 1; i >= 0; i-- {
+		currentPartition := partitionList[i]
+		if currentPartition > endPartition {
+			continue
+		} else if currentPartition < startPartition {
+			break
+		} else {
+			curPartitionPrefix := tablePrefix + currentPartition + "/"
+			itr.Seek([]byte(curPartitionPrefix))
+			stats.RowsVisitedCount += 1
+
+			if itr.ValidForPrefix([]byte(keyPrefix.String())) {
```

Comment:
Where are you updating the partition in keyPrefix?

Reply:
Good call, need to update KeyPrefix's partition based on current Partition